library/data-model/src/databaseEngine/check.ts

```diff
@@ -0,0 +1,826 @@
+/**
+ * A helper script which pulls and validates all records in a FAIMS deployment.
+ */
+import PouchDB from 'pouchdb';
+import * as fs from 'fs';
+import {ZodError} from 'zod';
+import {
+  getDocumentType,
+  recordDocumentSchema,
+  revisionDocumentSchema,
+  avpDocumentSchema,
+  attachmentDocumentSchema,
+  isRecordDocument,
+  hydratedRecordSchema,
+  DataDocument,
+} from './types';
+import {DataEngine} from './engine';
+
+// ============================================================================
+// Configuration
+// ============================================================================
+
+interface Config {
+  url: string;
+  username: string;
+  password: string;
+  testHydration: boolean;
+}
+
+function getConfig(): Config {
+  const args = process.argv.slice(2);
+
+  // Check for --test-hydration flag
+  const testHydrationIndex = args.indexOf('--test-hydration');
+  const testHydration = testHydrationIndex !== -1;
+
+  // Remove the flag from args if present
+  if (testHydrationIndex !== -1) {
+    args.splice(testHydrationIndex, 1);
+  }
+
+  if (args.length !== 3) {
+    console.error(
+      'Usage: tsx validate-database.ts <url> <username> <password> [--test-hydration]'
```

Comment:
That's not the name of the script, how are we going to use this if at all?  If it's going to be useful maybe make it a script in the api project.  Refactor so that the core functions could be called from web if this kind of audit is useful.

Reply:
Let's leave here for now as it's helpful for testing out the engine - we can then decide to keep or integrate properly into the CLI/API.